Backend/src/main/java/com/example/backend/companies/service/CompanyService.java

```diff
@@ -40,6 +40,22 @@ public List<Company> getAllCompanies(AppUser user) throws AuthenticationExceptio
         return companyRepository.findAll();
     }
 
+    private <T> T getValue(Optional<T> obj) throws EntityNotFoundException{
+        if(obj.isEmpty()) {
+            throw new EntityNotFoundException();
+        }
+        return obj.get();
+    }
+
+    public Boolean softLockCompany(Long id) throws EntityNotFoundException
+    {
+        Company company = getValue(companyRepository.findById(id));
+        boolean newSoftLock = company.getSoftLock() == null || !company.getSoftLock();
+        company.setSoftLock(newSoftLock);
+        companyRepository.save(company);
+        return newSoftLock;
+    }
+
```

Comment:
ovo rješenje mi se sviđa, ali nažalost nije dobro jer kad bacaš EntityNotFoundException iz servisa moraš mu predat poruku gdje piše koji entitet nije pronađen i pod kojim id-jem, npr "Project under id: <projectId> not found."